dae/dae/genomic_resources/genomic_scores.py

```diff
@@ -90,20 +91,51 @@ def _get_header(self):
         return self.table.get_column_names()
 
     def _fetch_lines(self, chrom, pos_begin, pos_end):
-        records = list(self.table.get_records_in_region(
-            chrom, pos_begin, pos_end))
+        records = self.table.get_records_in_region(
+            chrom, pos_begin, pos_end)
 
-        return [
-            ScoreLine(record, self.score_columns, self.special_columns)
-            for record in records
-        ]
+        for record in records:
+            yield ScoreLine(record, self.score_columns, self.special_columns)
 
     def get_all_chromosomes(self):
         return self.table.get_chromosomes()
 
     def get_all_scores(self):
         return list(self.score_columns)
 
+    def fetch_region(self, chrom: str, pos_begin: int, pos_end: int,
+                     scores: List[str]) \
+            -> dict[str, Generator[Number, None, None]]:
```

Comment:
I don't think this return type is a good idea. To have multiple independent generators would mean that we are going to iterate over all records multiple times.

I strongly prefer to have Generator[Dict[str, number], None, None].

Reply:
@lchorbadjiev fixed